cluster/kubernetes/resource/load.go

```diff
@@ -159,45 +158,8 @@ func ParseMultidoc(multidoc []byte, source string) (map[string]KubeManifest, err
 		}
 	}
 
-	if err := chunks.Err(); err != nil {
+	if err != io.EOF {
 		return objs, errors.Wrapf(err, "scanning multidoc from %q", source)
 	}
 	return objs, nil
 }
-
-// ---
-// Taken directly from https://github.com/kubernetes/apimachinery/blob/master/pkg/util/yaml/decoder.go.
-
-const yamlSeparator = "\n---"
-
-// splitYAMLDocument is a bufio.SplitFunc for splitting YAML streams into individual documents.
-func splitYAMLDocument(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
Although not related to this bugfix, another copy of this method can be found here: https://github.com/weaveworks/flux/blob/546ea7169eaede3da0b2193f09f263a35bec6474/cmd/fluxctl/save_cmd.go#L220

Reply:
yikes, that sucks

Reply:
Also totally not related to this PR, we probably want to include our GC labels and annotations here: https://github.com/weaveworks/flux/blob/546ea7169eaede3da0b2193f09f263a35bec6474/cmd/fluxctl/save_cmd.go#L102

Reply:
I will fix the first part. I am not sure about the second, can you create a separate issue for it?